src/parsing.jl

```diff
@@ -256,6 +256,81 @@ function Base.parse(::Type{ZonedDateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:int8]})
     TimeZones.unix2zdt(parse(Int64, pqv))
 end
 
+## intervals
+# iso_8601
+_DEFAULT_TYPE_MAP[:interval] = Dates.CompoundPeriod
+const INTERVAL_REGEX = Ref{Regex}()  # set at __init__
+
+function _interval_regex()
```

Comment:
The generated regex should probably have some test cases.

Reply:
Aren't the parse tests sufficient for that?

Reply:
Yes, but it'd be nice to have unit tests for just the behaviour of the regex created. Might make it easier for contributors to identify potential issues?